packages/compiler-cli/src/perform_watch.ts

```diff
@@ -88,7 +88,8 @@ export function createPerformWatchHost<CbEmitRes extends ts.EmitResult = ts.Emit
       const watcher = chokidar.watch(options.basePath, {
         // ignore .dotfiles, .js and .map files.
         // can't ignore other files as we e.g. want to recompile if an `.html` file changes as well.
-        ignored: /((^[\/\\])\..)|(\.js$)|(\.map$)|(\.metadata\.json|node_modules)/,
+        ignored: (path) =>
+          path.match(/((^[\/\\])\..)|(\.js$)|(\.map$)|(\.metadata\.json|node_modules)/) !== null,
```

Comment:
Since you don't need the matches array, you could switch this to test:

```suggestion
          /((^[\/\\])\..)|(\.js$)|(\.map$)|(\.metadata\.json|node_modules)/.test(path),
```

Reply:
Even better !